scripts/dependency-graph.js

```diff
@@ -1,189 +1,153 @@
+    "```mermaid",
+    "graph LR;",
     ...annotationsAll.sort(),
-    '```',
-    '',
-    '## npm dependency hierarchy',
-    '',
-    'These are the groups of dependencies in npm that depend on each other.',
-    'Each group depends on packages lower down the chain, nothing depends on',
-    'packages higher up the chain.',
-    '',
-    ` - ${hierarchyOurs.reverse().join('\n - ')}`,
-  ]
-
-  return fs.writeFile(join(CWD, 'DEPENDENCIES.md'), out.join('\n'))
+    "```",
+    "",
+    "## npm dependency hierarchy",
+    "",
+    "These are the groups of dependencies in npm that depend on each other.",
+    "Each group depends on packages lower down the chain, nothing depends on",
+    "packages higher up the chain.",
+    "",
+    ` - ${hierarchyOurs.reverse().join("\n - ")}`,
+  ];
+
+  await writeFile(join(__dirname, "DEPENDENCIES.md"), out.join("\n"));
 }
 
-const walk = function (tree, onlyOurs) {
-  const annotations = [] // mermaid dependency annotations
-  const dependedBy = {}
+function walk(tree, onlyOurs) {
+  const annotations = [];
+  const dependedBy = {};
 
-  iterate(tree, dependedBy, annotations, onlyOurs)
+  iterate(tree, dependedBy, annotations, onlyOurs);
 
-  const allDeps = new Set(Object.keys(dependedBy))
-  const foundDeps = new Set()
-  const hierarchy = []
+  const allDeps = new Set(Object.keys(dependedBy));
+  const foundDeps = new Set();
+  const hierarchy = [];
 
   if (onlyOurs) {
     while (allDeps.size) {
-      log.silly('SIZE', allDeps.size)
-      const level = []
+      const level = [];
 
       for (const dep of allDeps) {
-        log.silly(dep, '::', [...dependedBy[dep]].join(', '))
-        log.silly('-'.repeat(80))
-
         if (!dependedBy[dep].size) {
-          level.push(dep)
-          foundDeps.add(dep)
+          level.push(dep);
+          foundDeps.add(dep);
         }
       }
 
-      log.silly('LEVEL', level.length)
-      log.silly('FOUND', foundDeps.size)
-
       for (const dep of allDeps) {
         for (const found of foundDeps) {
-          allDeps.delete(found)
-          dependedBy[dep].delete(found)
+          allDeps.delete(found);
+          dependedBy[dep].delete(found);
         }
       }
 
-      log.silly('SIZE', allDeps.size)
-
       if (!level.length) {
-        const remaining = `Remaining deps: ${[...allDeps.keys()]}`
-        throw new Error(`Would do an infinite loop here, need to debug. ${remaining}`)
+        const remaining = `Remaining deps: ${[...allDeps.keys()]}`;
+        throw new Error(`Would do an infinite loop here, need to debug. ${remaining}`);
       }
 
-      hierarchy.push(level.join(', '))
-      log.silly('HIERARCHY', hierarchy.length)
-      log.silly('='.repeat(80))
+      hierarchy.push(level.join(", "));
     }
   }
 
-  return [annotations, hierarchy]
+  return [annotations, hierarchy];
 }
 
-const iterate = function (node, dependedBy, annotations, onlyOurs) {
+function iterate(node, dependedBy, annotations, onlyOurs) {
```

Comment:
`const foo = function` prevents reassignment of a function. We do this intentionally as a guard against this footgun.